renderdoc/driver/gl/gl_pixelhistory.cpp

```diff
@@ -151,72 +253,77 @@ bool PixelHistorySetupResources(WrappedOpenGL *driver, GLPixelHistoryResources &
     stencilFormat = driver->m_Textures[id].internalFormat;
   }
 
-  if(curStencil == curDepth)
-  {
-    driver->CreateTextureImage(resources.dsImage, depthFormat, eGL_NONE, eGL_NONE, eGL_TEXTURE_2D,
-                               2, numEvents, 1, 1, 1, 1);
-    driver->glFramebufferTexture(eGL_FRAMEBUFFER, eGL_DEPTH_STENCIL_ATTACHMENT, resources.dsImage, 0);
-  }
-  else
-  {
-    if(curDepth != 0)
-    {
-      driver->CreateTextureImage(resources.depthImage, depthFormat, eGL_NONE, eGL_NONE,
-                                 eGL_TEXTURE_2D, 2, numEvents, 1, 1, 1, 1);
-      driver->glFramebufferTexture(eGL_FRAMEBUFFER, eGL_DEPTH_ATTACHMENT, resources.depthImage, 0);
-    }
-    if(curStencil != 0)
-    {
-      driver->CreateTextureImage(resources.stencilImage, stencilFormat, eGL_NONE, eGL_NONE,
-                                 eGL_TEXTURE_2D, 2, numEvents, 1, 1, 1, 1);
-      driver->glFramebufferTexture(eGL_FRAMEBUFFER, eGL_STENCIL_ATTACHMENT, resources.stencilImage,
-                                   0);
-    }
-  }
-
-  // Allocate a framebuffer that will render to the textures
-  driver->glGenFramebuffers(1, &resources.frameBuffer);
-  driver->glBindFramebuffer(eGL_FRAMEBUFFER, resources.frameBuffer);
+  return getCopyFramebuffer(driver, copyFramebuffers, numSamples, numEvents, depthFormat,
+                            stencilFormat);
+}
 
-  // Allocate a texture for the pixel history colour values
-  driver->glGenTextures(1, &resources.colorImage);
-  driver->glBindTexture(eGL_TEXTURE_2D, resources.colorImage);
-  driver->CreateTextureImage(resources.colorImage, eGL_RGBA32F, eGL_NONE, eGL_NONE, eGL_TEXTURE_2D,
-                             2, numEvents, 1, 1, 1, 1);
-  driver->glFramebufferTexture(eGL_FRAMEBUFFER, eGL_COLOR_ATTACHMENT0, resources.colorImage, 0);
+GLbitfield getFramebufferCopyMask(WrappedOpenGL *driver)
+{
+  GLuint curDepth;
+  GLuint curStencil;
 
-  // Allocate a texture for the pixel history depth/stencil values
-  driver->glGenTextures(1, &resources.dsImage);
-  driver->glBindTexture(eGL_TEXTURE_2D, resources.dsImage);
+  driver->glGetFramebufferAttachmentParameteriv(eGL_DRAW_FRAMEBUFFER, eGL_DEPTH_ATTACHMENT,
+                                                eGL_FRAMEBUFFER_ATTACHMENT_OBJECT_NAME,
+                                                (GLint *)&curDepth);
 
-  GLResource depthResource;
+  driver->glGetFramebufferAttachmentParameteriv(eGL_DRAW_FRAMEBUFFER, eGL_STENCIL_ATTACHMENT,
+                                                eGL_FRAMEBUFFER_ATTACHMENT_OBJECT_NAME,
+                                                (GLint *)&curStencil);
 
-  driver->CreateTextureImage(resources.dsImage, eGL_DEPTH24_STENCIL8, eGL_NONE, eGL_NONE,
-                             eGL_TEXTURE_2D, 2, numEvents, 1, 1, 1, 1);
-  driver->glFramebufferTexture(eGL_FRAMEBUFFER, eGL_DEPTH_STENCIL_ATTACHMENT, resources.dsImage, 0);
+  GLbitfield mask = eGL_COLOR_BUFFER_BIT;
```

Comment:
This might be fine in practice if GL allows it, I'm not sure off the top of my head, but will this work if the framebuffer doesn't have a colour attachment and only has depth? e.g. for shadow rendering or similar?